src/const.jl

```diff
@@ -0,0 +1,11 @@
+struct Const{T} <: Function
```

Comment:
As I said on Zulip, this exists as `Returns` in [1.7](https://github.com/JuliaLang/julia/pull/39794).  Maybe use some compat tricks to use that instead.

Also, the [constructor](https://github.com/JuliaLang/julia/pull/39794/files#diff-12e7a6522633012a408b1bdee7639e8cb722617fe1a8ed6a3881bf4ad1ebdbbdR941) over there is different, maybe that's relevant.

Reply:
Hey @phipsgabler, sorry I had missed this. I agree, let's move toward `Returns` and `Compat`